logging: compiler warning when compiling `log_multidomain_backend.c`

**Is your enhancement proposal related to a problem? Please describe.**
Under certain configurations, the following warning appears in the terminal during compilation, and might worth taking a closer look:

```
[134/158] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/backends/log_multidomain_backend.c.obj
/home/ycsin/zephyrproject/zephyr/subsys/logging/backends/log_multidomain_backend.c: In function ‘process’:
/home/ycsin/zephyrproject/zephyr/subsys/logging/backends/log_multidomain_backend.c:41:57: warning: taking address of packed member of ‘union log_multidomain_msg_data’ may result in an unaligned pointer value [-Waddress-of-packed-member]
   41 |         struct log_msg *out_log_msg = (struct log_msg *)out_msg->data.log_msg.data;
      |                                                         ^~~~~~~
```

**Describe the solution you'd like**
There shouldn't be any warnings during compilation.

**Describe alternatives you've considered**
Ignore the issue.

--- a/subsys/logging/backends/log_multidomain_backend.c
+++ b/subsys/logging/backends/log_multidomain_backend.c
@@ -32,13 +32,16 @@ static void process(const struct log_backend *const backend,
 		return;
 	}
 
-	/* Need to ensure that package is aligned to a pointer size. */
+	/* Need to ensure that package is aligned to a pointer size even though
+	 * it is in the packed structured.
+	 */
 	uint32_t msg_len = Z_LOG_MSG_LEN(fsc_plen, dlen);
 	uint8_t buf[msg_len + sizeof(void *)] __aligned(sizeof(void *));
 	size_t msg_offset = offsetof(struct log_multidomain_msg, data);
 	struct log_multidomain_msg *out_msg =
 		(struct log_multidomain_msg *)&buf[sizeof(void *) - msg_offset];
-	struct log_msg *out_log_msg = (struct log_msg *)out_msg->data.log_msg.data;
+	uintptr_t out_log_msg_ptr = (uintptr_t)out_msg->data.log_msg.data;
+	struct log_msg *out_log_msg = (struct log_msg *)out_log_msg_ptr;
 
 	/* Set ipc message id. */
 	out_msg->id = Z_LOG_MULTIDOMAIN_ID_MSG;

